File: java-utils/src/main/java/com/cds/common/metrics/source/JvmSource.java
package com.cds.common.metrics.source;

import com.codahale.metrics.MetricRegistry;
import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
import com.codahale.metrics.jvm.MemoryUsageGaugeSet;

import javax.annotation.concurrent.ThreadSafe;

/**
 * A Source which collects JVM metrics, including JVM memory usage, GC counts, GC times, etc.
 */
@ThreadSafe
public class JvmSource implements Source {
  private static final String JVM_SOURCE_NAME = "jvm";
  private final MetricRegistry mMetricRegistry;

  /**
   * Creates a new {@link JvmSource} and register all JVM metrics.
   */
  public JvmSource() {
    mMetricRegistry = new MetricRegistry();
    mMetricRegistry.registerAll(new GarbageCollectorMetricSet());
    mMetricRegistry.registerAll(new MemoryUsageGaugeSet());
  }

  @Override
  public String getName() {
    return JVM_SOURCE_NAME;
  }

  @Override
  public MetricRegistry getMetricRegistry() {
    return mMetricRegistry;
  }
}

File: scala-utils/src/main/scala/com/cds/utils/common/metrics/source/JvmSource.scala
package com.cds.utils.common.metrics.source

import com.codahale.metrics.MetricRegistry
import com.codahale.metrics.jvm.{GarbageCollectorMetricSet, MemoryUsageGaugeSet}

/**
  * Created by chendongsheng5 on 2017/6/14.
  */
class JvmSource extends Source {
  override val sourceName = "jvm"
  override val metricRegistry = new MetricRegistry()

  metricRegistry.registerAll(new GarbageCollectorMetricSet)
  metricRegistry.registerAll(new MemoryUsageGaugeSet)
}
